cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/validator/AzureImageFormatValidator.java

```diff
@@ -49,12 +49,18 @@ public void validate(AuthenticatedContext ac, CloudStack cloudStack) {
         Image image = cloudStack.getImage();
         String imageUri = image.getImageName();
 
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
         if (isVhdImageFormat(image)) {
+            LOGGER.debug("Checking presence of 'Only Azure Marketplace Images Enabled' entitlement for VHD image {}", imageUri);
+            if (entitlementService.azureOnlyMarketplaceImagesEnabled(accountId)) {
+                String errorMessage = String.format("Your image %s seems to be a VHD image, but only Azure Marketplace images allowed in your account! "
+                        + "If you would like to use it please open Cloudera support ticket to enable this capability!", imageUri);
+                LOGGER.warn(errorMessage);
+                throw new CloudConnectorException(errorMessage);
```

Comment:
could you pls externalise this as a private method?

Reply:
Looks like we are validating this as part of stack creation/provisioning. In my view, We should have this validation while determining the image. Currently, if Image id is provided in the request, VHD image can be determined even if Azure MP entitlement is assigned. Please refer to findImage API of FreeIpaImageProvider class.

Reply:
We did not update image filtering logic to check entitlement if image id is provided in the request so image filtering logic will return vhd image even if Azure Only MP entitlement is assigned. Looks like assumption is that this logic will always be executed before image filtering for all use cases. Please check.